pkg/logql/log/parser_hints.go

```diff
@@ -95,19 +91,25 @@ func (p *Hints) NoLabels() bool {
 }
 
 func (p *Hints) RecordExtracted(key string) {
-	for _, l := range p.requiredLabels {
-		if l == key {
-			p.extracted = append(p.extracted, key)
-			return
-		}
-	}
+	p.extracted = append(p.extracted, key)
 }
 
 func (p *Hints) AllRequiredExtracted() bool {
-	if len(p.requiredLabels) == 0 {
+	if len(p.requiredLabels) == 0 || len(p.extracted) < len(p.requiredLabels) {
 		return false
 	}
-	return len(p.extracted) == len(p.requiredLabels)
+
+	found := map[string]interface{}{}
+	for _, e := range p.extracted {
+		for _, l := range p.requiredLabels {
+			if e == l {
+				found[l] = nil
+				break
+			}
+		}
+	}
+
+	return len(p.requiredLabels) == len(found)
```

Comment:
this is needed because previously, `RecordExtracted` was only recording required fields that were extracted. As a result, it was previously acceptable to just test the length of the 2 slices against each other. However, now that we're recording _all_ extracted labels, we have to actually compare extracted to required. I'm using a map to prevent duplicate extractions from causing an incorrect result here.